src/mapml/layers/MapLayer.js

```diff
@@ -521,19 +521,54 @@ export var MapMLLayer = L.Layer.extend({
         opacity.setAttribute('step','0.1');
         opacity.value = this._container.style.opacity || '1.0';
 
-        L.DomEvent.on(opacity,'change', this._changeOpacity, this);
-        L.DomEvent.on(name,'dragstart', function(event) {
-            // will have to figure out how to drag and drop a whole element
-            // with its contents in the case where the <layer->content</layer-> 
-            // has no src but does have inline content.  
-            // Should be do-able, I think.
-            if (this._href) {
-              event.dataTransfer.setData("text/uri-list",this._href);
-              // Why use a second .setData("text/plain"...) ? This is very important:
-              // See https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API/Recommended_drag_types#link
-              event.dataTransfer.setData("text/plain", this._href); 
+        fieldset.setAttribute("aria-grabbed", "false");
+        fieldset.draggable = true;
+
+        fieldset.onmousedown = (e) => {
+          e.target.closest("fieldset").draggable = e.target.tagName.toLowerCase() !== "input";
+        };
+
+        fieldset.onmouseup = (e) => {
+          e.target.closest("fieldset").draggable = true;
+        };
+
+        fieldset.ondrag = (e) => {
+          let control = e.target,
+              controls = e.target.parentNode,
+              x = e.clientX, y = e.clientY,
+              elementAt = root.elementFromPoint(x, y),
+              swapControl = !elementAt || !elementAt.closest("fieldset") || elementAt.closest("fieldset").draggable === false ? control : elementAt.closest("fieldset");
+          control.classList.add("drag-active");
+          control.setAttribute("aria-grabbed", 'true');
+          control.setAttribute("aria-dropeffect", "move");
+          if(swapControl && controls === swapControl.parentNode){
+            swapControl = swapControl !== control.nextSibling? swapControl : swapControl.nextSibling;
+            controls.insertBefore(control, swapControl);
+          }
+        };
+        fieldset.ondragend = (e) => {
+          e.target.classList.remove("drag-active");
+          e.target.setAttribute("aria-grabbed", "false");
+          e.target.removeAttribute("aria-dropeffect");
+          let controls = e.target.parentNode.children,
+              layers = map.getPane("overlayPane").children,
+              zIndex = 1;
+          for(let control of controls){
+            let layerEl = control.querySelector("span").layer._layerEl;
+            layerEl.setAttribute("moving","");
```

Comment:
`moving` as opposed to e.g. `data-moving` has the same concern as https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/191. Just an FYI (and also to create a reference in that issue, in case we decide to follow the recommendation we know what to change).

Reply:
You suggest using `data-moving` instead? Either works, I just need a way to prevent making unneeded requests when moving the layer element.

Reply:
Yes, authors are encouraged to used [`data-`](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) prefixed attributes, this allows the web platform to be extensible by not breaking sites when adding new (non-prefixed) attributes.